### DISCUSSION TITLE
代理服务器那端一般用什么？

### DISCUSSION THREAD

**Discussion by ccdd4ever** [xjasonlyu/tun2socks#121] [Q&A]
tun2socks按我理解是运行在客户端，服务器那一段一般用什么软件？nginx？

**Comment by xjasonlyu** [comment]
nginx不是代理服务器吧。

服务端可以使用任何支持HTTP/SOCKS4/SOCKS5/Shadowsocks的服务端，如Squid，BurpSuite，Charles，Clash，V2Ray，Shadowsocks-libev……

**Reply by ccdd4ever** [reply]
请问下这里面性能和稳定性最成熟的是哪一个。nginx按我理解也可以用stream模式接TCP流量，然后用lua解析socks5协议吧？不过这样做似乎不是很优雅

**Reply by xjasonlyu** [reply]
这当然是看你的需求场景的，不同后端需求场景都是不同的。

**Reply by ccdd4ever** [reply]
场景是作为代理服务器/网关。接入的用户数会比较多，用于替换VPN。这种场景下比较适合的方案是什么呢？

**Reply by xjasonlyu** [reply]
那你就是需要一个多用户管理的代理服务端，如果不需要UDP的话可以考虑Squid

**Reply by ccdd4ever** [reply]
感谢，我了解下Squid.  最近也在看Xray和v2ray，这两个产品适合多用户的场景嘛？

**Reply by xjasonlyu** [reply]
确实可以，用户管理自己用grpc调用就行

**Reply by ccdd4ever** [reply]
好的，非常感谢大佬。
